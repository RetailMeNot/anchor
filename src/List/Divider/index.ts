export { Divider } from './Divider.component';
