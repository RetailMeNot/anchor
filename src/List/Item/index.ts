export { Item } from './Item.component';
