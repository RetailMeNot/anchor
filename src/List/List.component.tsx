// REACT
import * as React from 'react';
// VENDOR
import classNames from 'classnames';
import styled from 'styled-components';
// COMPONENTS
// THEME
import { fonts, colors, sizes } from '../theme';

interface ListProps {
  className?: string;
  children?: any;
  title?: string;
  hidden?: boolean;
}

export const StyledList = styled.div`
  .title {
    margin: 0;
    padding: ${sizes.padding.md};
    font-family: ${fonts.fontFamily};
    font-size: .75rem;
    font-weight: 600;
    color: ${colors.neutrals.ash.dark};
    line-height: .75rem;
  }
`;

const DefaultProps: ListProps = {
};

export const List: React.SFC<ListProps> = ({
  className,
  children,
  title,
...props
}: ListProps = DefaultProps): React.ReactElement<any> => (
  <StyledList className={classNames(className)} {...props}>
    {title && <p className="title">{title}</p>}
    {children}
  </StyledList>
);

List.defaultProps = DefaultProps;

export default List;
