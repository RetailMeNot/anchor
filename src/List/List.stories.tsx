// REACT
import * as React from 'react';
// STORYBOOK
import { storiesOf } from '@storybook/react';
// VENDOR
import styled from 'styled-components';
// COMPONENTS
import { List, Item } from './';
// THEME

const StyledStory = styled.div`
  padding: 2rem 5rem;
`;

storiesOf('Components/List', module)
  .add('Default', () => (
      <StyledStory>
          <p>List</p>
          <List>
            <Item>Item 1</Item>
            <Item>Item 3</Item>
            <Item>Item 4</Item>
            <Item>Item 5</Item>
            <Item>Item 6</Item>
            <Item>Item 7</Item>
          </List>
      </StyledStory>
  ));
