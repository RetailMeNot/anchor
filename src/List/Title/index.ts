export { Title } from './Title.component';
