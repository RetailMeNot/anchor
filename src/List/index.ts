export { List, ListItemType } from './List.component';
export { Item } from './Item/Item.component';
export { Title } from './Title/Title.component';
export { Divider } from './Divider/Divider.component';
