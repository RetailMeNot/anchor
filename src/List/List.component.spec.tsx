// REACT
import * as React from 'react';
import * as renderer from 'react-test-renderer';
// COMPONENT
import List from './List.component';
// ENZYME
import { shallow, mount } from 'enzyme';

// TEST SETUP
const subject = <List />;
const wrapper = mount(subject);
const component = shallow(subject);

describe('Component: List', () => {
    it('should be defined', () => {
        /* tslint:disable */
        console.log(subject);
        console.log(wrapper.debug());
        console.log(component.debug());

        /* tslint:enable */
        expect(subject).toBeDefined();
        expect(wrapper).toBeDefined();
        expect(component).toBeDefined();
        const tree = renderer.create(subject).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
