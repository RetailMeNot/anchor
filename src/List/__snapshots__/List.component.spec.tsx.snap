// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: List should be defined 1`] = `
<div
  className="anchor-list sc-bxivhb cgDDEg"
/>
`;
